File: Python/RemoveSubfoldersFromtheFilesystem.py
class Solution(object):
    def removeSubfolders(self, folder):
        folder.sort()
        i = 0

        while i < len(folder):
            subfolder = folder[i]
            seek = i+1

            while seek < len(folder) and folder[seek].startswith(subfolder + '/'):
                seek += 1
            
            folder = folder[:i + 1] + folder[seek:]
            i+=1

        return folder

File: C++/RemoveSub-FoldersfromtheFilesystem.cpp
class Solution {
public:
    vector<string> removeSubfolders(vector<string>& folder) {
        sort(folder.begin(), folder.end());
        vector<string> ret;

        for (string str: folder) {
            if (ret.empty() || str.substr(0, ret.back().size()) != ret.back() || str[ret.back().size()] != '/') {
                ret.push_back(str);
            }
        }

        return ret;
    }
};
